de_web_plugin.cpp

```diff
@@ -9013,6 +9020,28 @@ void DeRestPluginPrivate::updateSensorNode(const deCONZ::NodeEvent &event)
                                     updated = true;
                                 }
                             }
+                            else if (ia->id() == 0x050f) // Apparent power
+                            {
+                                if (updateType != NodeValue::UpdateInvalid)
+                                {
+                                    i->setZclValue(updateType, event.endpoint(), event.clusterId(), ia->id(), ia->numericValue());
+                                    pushZclValueDb(event.node()->address().ext(), event.endpoint(), event.clusterId(), ia->id(), ia->numericValue().u16);
+                                }
+
+                                quint16 power = ia->numericValue().u16;
+                                ResourceItem *item = i->item(RStatePower);
+
+                                if (item && power != -32768)
```

Comment:
Compiler complains:

```
de_web_plugin.cpp:9035:51: warning: comparison is always true due to limited range of data type [-Wtype-limits]
 9035 |                                 if (item && power != -32768)
```

It's an `uint16` so it can't be lower than 0, where does the -32768 come from?

Reply:
From the wrong data type